metis-core/metis-core-rest/src/main/java/eu/europeana/metis/core/rest/exception/RestResponseExceptionHandler.java

```diff
@@ -103,4 +104,29 @@ public StructuredExceptionWrapper handleMessageNotReadable(Exception exception,
     return new StructuredExceptionWrapper(
         "Request not readable.\n" + exception.getMessage());
   }
+
+  /**
+   * Handler for specific classes to overwrite behaviour
+   * @param exception the exception thrown
+   * @param response the response that should be updated
+   * @return {@link StructuredExceptionWrapper} a json friendly class that contains the error message for the client
+   */
+  @ExceptionHandler(value = MissingRequestHeaderException.class)
+  @ResponseBody
+  public StructuredExceptionWrapper handleMissingRequestHeaderException(Exception exception,
+      HttpServletResponse response) {
+
+    String header = ((MissingRequestHeaderException) exception).getHeaderName();
```

Comment:
You can define the exception parameter of type MissingRequestHeaderException and avoid the cast.
Also the method is quite simple and descriptive, so you could directly use exception.getHeaderName() on the if statement.

Reply:
Good point, will do that